tests/manual/test_memory_reporting.py

```
@@ -0,0 +1,293 @@
+    if abs(b) >= 1024:
+        return f"{b / 1024:.2f} KB"
+    return f"{b} B"
+
+
+def test_memory_reporting_accuracy():
+    """
+    Load a snapshot, measure memory, remove components one by one,
+    and compare reported vs actual memory freed.
+    """
+    # Verify server is up
+    resp = requests.get(f"{BASE_URL}/collections")
```

Ruff (0.15.9)

[error] Probable use of `requests` call without timeout

(S113)